### DISCUSSION TITLE
Why std::filesystem::exists returns false for c:\hiberfil.sys

### DISCUSSION THREAD

**Discussion by s-kipnis** [microsoft/STL#2365] [Q&A] [answered]
This behavior is at least strange. The file does exist, but has quite limited permissions.
Is it bug or side effect of Win32 API?

**Comment by viccpp** [comment]
May be because you write `"c:\hiberfil.sys"` instead of `"c:\\hiberfil.sys"`?

**Comment by s-kipnis** [comment]
Nope. You may try.

Approximately:
```
#include <filesystem>
#include <iostream>
#include <system_error>

int main(){
   std::error_code ec;
   std::cout << std::filesystem::exists("c:\\hiberfil.sys", ec );
   return 0;
}
```
error_code is set to 32 with message() ~ "no permission ... the file in use"

**Reply by viccpp** [reply]
Confirm. Looks like a bug

**Reply by viccpp** [reply]
Interestingly. `GetFileAttributesW()` returns `ERROR_SHARING_VIOLATION`. We can treat this result as path existence but we cannot distinguish regular files from directories. But file attributes can be obtained without such error using `FindFirstFile()`

**Comment by s-kipnis** [comment]
Yes, you are right: we can treat ERROR_SHARING_VIOLATION as path existence.
We must use next workaround:
```
for (auto const &dir_entry : fs::directory_iterator{file_path.parent_path()}) {
//everything is ok also for files with ERROR_SHARING_VIOLATION
}
```

**Reply by viccpp** [reply]
Anyways `std::filesystem::exists()` must work as user expects - just return `true` without any error

**Comment by StephanTLavavej** [comment, marked as answer]
Thanks - this sounds like a bug, so we'll convert this into an issue!

**Reply by StephanTLavavej** [reply]
@strega-nil-ms fixed this with #2715 in VS 2022 17.3 Preview 3.
